src/main.rs

```diff
@@ -59,12 +60,32 @@ use cli::generate_cli;
 use decompilation::*;
 use static_analysis::*;
 use results::*;
+use errors::*;
 pub use config::Config;
 pub use utils::*;
 
 static BANNER: &'static str = include_str!("banner.txt");
 
 fn main() {
+    if let Err(e) = run() {
+
+        for e in e.iter().skip(1) {
```

Comment:
This should probably be more elaborated, for example, using `e.description()` should give more details of the issue. Also, you are discarding the first error, which shouldn't happen. Not sure how this glues together with printing warnings and errors during the analysis. If we are returning an error for chaining, nothing should be printed until we get here, if we print something inside run(), must be recoverable (e.g. warnings and debug messages).